emma-language/src/main/scala/org/emmalanguage/api/DataBag.scala

```diff
@@ -311,6 +316,34 @@ trait DataBag[A] extends Serializable {
           }
       })._1
   }
+
+  // -----------------------------------------------------
+  // equals, hashCode and toString
+  // -----------------------------------------------------
+
+  override def equals(o: Any) = o match {
+    case that: DataBag[A] => {
+      val thisVals = fetch()
+      val thatVals = that.fetch()
+      (thisVals.size == thatVals.size) &&
```

Comment:
I removed it deliberately for a number of reasons:

1. None of the backend API have native support for difference (and we didn't support translating `minus` in the old backend).
- It was never used in any of the examples.